internal/cli/atlas/accesslogs/list.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslogs
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	success      = "success"
+	fail         = "fail"
+	listTemplate = `HOSTNAME	CLUSTER NAME	LOG LINE {{range .AccessLogs}}
+{{if .Hostname}}.Hostname {{else}}N/A{{end}}{{.Hostname}}	{{if .ClusterName}}.ClusterName {{else}}N/A{{end}}	{{.LogLine}}{{end}}
+`
+	missingClusterNameHostnameErrorMessage = "one between --clusterName and --hostname must be set"
+	invalidValueAuthResultErrorMessage     = `--authResult must be set to "%s" or "%s"`
+)
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	hostname    string
+	clusterName string
+	start       string
+	end         string
+	nLogs       int
+	ipAddresses string
+	authResult  string
+	store       store.AccessLogsLister
+}
+
+func (opts *ListOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+func (opts *ListOpts) Run() error {
+	if opts.clusterName != "" {
+		r, err := opts.store.AccessLogsByClusterName(opts.ConfigProjectID(), opts.clusterName, opts.newAccessLogOptions())
+		if err != nil {
+			return err
+		}
+
+		return opts.Print(r)
+	}
+
+	r, err := opts.store.AccessLogsByHostname(opts.ConfigProjectID(), opts.hostname, opts.newAccessLogOptions())
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *ListOpts) newAccessLogOptions() *atlas.AccessLogOptions {
+	authResult := false
+	if strings.EqualFold(opts.authResult, success) {
+		authResult = true
+	}
+
+	return &atlas.AccessLogOptions{
+		Start:      opts.start,
+		End:        opts.end,
+		NLogs:      opts.nLogs,
+		IPAddress:  opts.ipAddresses,
+		AuthResult: &authResult,
+	}
```

Comment:
I left a similar comment on the client PR but I believe empty(nil) `AuthResult` is valid and should bring both failed and successful access logs, can you confirm this and if that's the case default `AuthResult` to nil and not `false`

This also implies that the flag accepts empty("") as valid

Reply:
no, when authResult is set to nil, it returns only the successful access logs. See
```json

~/workspace/mongoCLI git:(CLOUDP-97430) ✗ ./bin/mongocli atlas accessLogs list --clusterName Cluster0 --nLog 1 --authResult success -o json
{
  "accessLogs": [
    {
      "groupId": "5e4e593f70dfbf1010295836",
      "hostname": "atlas-avx159-shard-00-02.ajlj3.mongodb-dev.net",
      "ipAddress": "77.107.233.162",
      "authResult": true,
      "logLine": "{\"t\":{\"$date\":\"2021-09-08T10:30:17.862+00:00\"},\"s\":\"I\",  \"c\":\"ACCESS\",   \"id\":20250,   \"ctx\":\"conn1653\",\"msg\":\"Authentication succeeded\",\"attr\":{\"mechanism\":\"SCRAM-SHA-1\",\"speculative\":false,\"principalName\":\"test\",\"authenticationDatabase\":\"admin\",\"remote\":\"77.107.233.162:53767\",\"extraInfo\":{}}}",
      "timestamp": "Wed Sep 08 10:30:17 GMT 2021",
      "username": "test",
      "authSource": "admin"
    }
  ]
}
➜  ~/workspace/mongoCLI git:(CLOUDP-97430) ✗ ./bin/mongocli atlas accessLogs list --clusterName Cluster0 --nLog 1 --authResult fail -o json
{
  "accessLogs": [
    {
      "groupId": "5e4e593f70dfbf1010295836",
      "hostname": "atlas-avx159-shard-00-02.ajlj3.mongodb-dev.net",
      "ipAddress": "77.107.233.162",
      "authResult": false,
      "logLine": "{\"t\":{\"$date\":\"2021-09-08T10:30:17.676+00:00\"},\"s\":\"I\",  \"c\":\"ACCESS\",   \"id\":20249,   \"ctx\":\"conn1653\",\"msg\":\"Authentication failed\",\"attr\":{\"mechanism\":\"SCRAM-SHA-256\",\"speculative\":true,\"principalName\":\"test\",\"authenticationDatabase\":\"admin\",\"remote\":\"77.107.233.162:53767\",\"extraInfo\":{},\"error\":\"BadValue: SCRAM-SHA-256 authentication is disabled\"}}",
      "timestamp": "Wed Sep 08 10:30:17 GMT 2021",
      "username": "test",
      "failureReason": "BadValue: SCRAM-SHA-256 authentication is disabled",
      "authSource": "admin"
    }
  ]
}
➜  ~/workspace/mongoCLI git:(CLOUDP-97430) ✗ ./bin/mongocli atlas accessLogs list --clusterName Cluster0 --nLog 1 -o json
{
  "accessLogs": [
    {
      "groupId": "5e4e593f70dfbf1010295836",
      "hostname": "atlas-avx159-shard-00-02.ajlj3.mongodb-dev.net",
      "ipAddress": "77.107.233.162",
      "authResult": true,
      "logLine": "{\"t\":{\"$date\":\"2021-09-08T10:30:17.862+00:00\"},\"s\":\"I\",  \"c\":\"ACCESS\",   \"id\":20250,   \"ctx\":\"conn1653\",\"msg\":\"Authentication succeeded\",\"attr\":{\"mechanism\":\"SCRAM-SHA-1\",\"speculative\":false,\"principalName\":\"test\",\"authenticationDatabase\":\"admin\",\"remote\":\"77.107.233.162:53767\",\"extraInfo\":{}}}",
      "timestamp": "Wed Sep 08 10:30:17 GMT 2021",
      "username": "test",
      "authSource": "admin"
    }
  ]
}
```

Reply:
I will default to `true` instead of `false`

Reply:
sorry to be a drag here just want to be extra sure of this, is this the same with `curl`? and if yes then I'd be in favour of not making it `--flag=<fail|success>` but then making it just `--failed` which reduces the validations you have to do and follows the API experience a bit closer

Reply:
no, I haven’t used `curl`. I will check now and update this thread. Makes sense to have only the flag `--failed` in case the behaviour is the same with `curl`

Reply:
never mind my previous comment, I didn’t notice that I was providing the flag `--nLog 1` so the command was showing only the first entry in the log. You are right, when we provide `nil` the endpoints return both. Thanks for being a drag on this!  I will update it asap